institutions-ui/src/app/institutions-editor/institutions-editor.component.ts

```diff
@@ -35,10 +35,25 @@ export class InstitutionsEditorComponent implements OnInit {
   ngOnInit(): void {
     if(this.institutionId) {
       this.instService.getInstitutionDetails(this.institutionId).subscribe(inst=>this.institution=inst)
+    } else {
+      this.instService.getNextInstitutionId().subscribe(({next_id})=>this.institution.id = next_id)
     }
   }
 
+  extractErrorMessage(err: any) {
+    // Try to extract a meaningful error message out of the object returned by the server
+    return err?.error?.detail[0]?.msg || err?.error?.detail || err.message 
+  }
+
+  sanitizeFormData() {
+    // clean up form data prior to submission
+    let {name, id, ror_id } = this.institution
+    this.institution = { name: name.trim(), id: id.trim(), ror_id: ror_id?.trim() }
```

Comment:
@brianhlin I believe the issue you saw with duplicate names was a case of one duplicate having a trailing space. This should sanitize leading/trailing whitespace from form data before submitting to the server, where the DB unique constraints will catch duplicates

Reply:
Nice catch!